### PR TITLE
feat(wiki): markdown editor

### PR DESCRIPTION
**Pull request by AyshaHakeem**
- Add a patch to convert 'content' field's value in existing Wiki Page records from Rich Text (HTML) to Markdown
- Update the 'content' field's type in the Wiki Page Doctype to 'Markdown Editor’
- Convert existing editor to markdown in the frontend

### CODE REVIEW

File: wiki/public/js/editor.js [hunk 1]
```diff
@@ -1,63 +1,83 @@
...
+const editorContainer = document.getElementById("wiki-editor");
+const previewToggleBtn = $("#toggle-btn");
+const wikiTitleInput = $(".wiki-title-input");
+const saveWikiPageBtn = document.querySelector(
+  '[data-wiki-button="saveWikiPage"]'
+);
+const draftWikiPageBtn = document.querySelector(
+  '[data-wiki-button="draftWikiPage"]'
+);
+let show_preview = false;
```
Comment (shariquerik) [inline wiki/public/js/editor.js:14]: Since all variables are in camelCase stick with camelCase

File: wiki/public/js/render_wiki.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ function toggleEditor() {
   $(".wiki-content").toggleClass("hide");
   $(".wiki-page-meta").toggleClass("hide");
   $(".wiki-footer").toggleClass("hide");
-  $(".wiki-edit-control-btn").toggleClass("hide");
+  // $(".wiki-edit-control-btn").toggleClass("hide");
```
Comment (shariquerik) [inline wiki/public/js/render_wiki.js:49]: remove it if not required

File: wiki/public/scss/edit_wiki.scss [hunk 3]
```diff
@@ -1,610 +1,633 @@
 @import "frappe/public/scss/website/variables";
```
Comment (shariquerik) [inline wiki/public/scss/edit_wiki.scss:2]: Remove unused styles if there are any

**Review by shariquerik — CHANGES_REQUESTED** [review]

**Comment by shariquerik** [comment]
@AyshaHakeem Can you also fix the failing cypress test

**Comment by shariquerik** [comment]
Suggestion: Always try to push in multiple and small commits instead of pushing everything in one commit. It will be easier for reviewer to read your PR.

**Comment by shariquerik** [comment]
I was testing and tried to create CRM's Introduction page which works perfectly. But I have added a video tag is not visible in edit mode after saving. It is visible in desk but not in portal

image


Desk:
image

Portal:
image

File: wiki/public/js/editor.js [hunk 4]
```diff
@@ -1,63 +1,83 @@
...
+
+let editor = Ace.edit(editorContainer, {
+  mode: "ace/mode/markdown",
+  placeholder: "Wiki Content",
```
Comment (shariquerik) [inline wiki/public/js/editor.js:18]:
```suggestion
  placeholder: "Write your content here...",
```

**Review by shariquerik — CHANGES_REQUESTED** [review]

**Comment by AyshaHakeem** [comment]
Thank you Shariq for reviewing! @shariquerik 
I have made the suggested changes.

https://github.com/user-attachments/assets/87390a28-2176-4595-b71d-e5220a1fbebc

**Comment by kittiu** [comment]
Hello, this is quite urgent. My team is working on documentation and more comforatble with rich text.
Is it configurable to use Rich Text?

**Comment by kittiu** [comment]
> > Note: Just wondering why are we changing the editor?
> 
> Markdown is a standard way of writing docs hence the switch. It take less than an hour to learn Markdown.

OK 👌 just want to get confirmed. Thanks!

**Comment by kittiu** [comment]
@shariquerik we do have problem with wiki writing pane for our local language.

image

This is actually happen in any coding syle pane (i.e., system console) but not rich text. In the pic you will see the cursor position not get along with the real display.

Any hint to solve this?

**Comment by kittiu** [comment]
May be the problem with ace editor.

**Pull request merged**